web-server/src/controllers/tasks/helpers/createScript.ts

```diff
@@ -0,0 +1,14 @@
+import { agent as request } from 'supertest';
+
+import { createFile } from './createFile';
+import { ScriptPayload } from './payloads';
+
+import { app } from '../../../index';
+
+export const createScript = async (data: ScriptPayload) => {
+  const { file, languageCode, runtimeArgs } = data;
+
+  const createdFile = await createFile(file);
+  const res = await request(app).post('/v1/scripts').send({ file: createdFile, languageCode, runtimeArgs });
```

Comment:
Oh no. Please don't call an endpoint internally.

Just call the appropriate function directly! This allows you to have nice things like memory locality, database transactions, and a non-rats-nest of recursive middleware shenanigans.